shared/cluster.go

```diff
@@ -409,3 +409,43 @@ func WriteDataPod(namespace string) (string, error) {
 
 	return RunCommandHost(cmd)
 }
+
+// Get service name. Used to work with stop/start k3s/rke2 services
+func getServiceName(product string, serviceType string) string {
+	var serviceName string
+	if product == "k3s" && serviceType == "server" {
+		serviceName = product // k3s
+	} else { // k3s-agent, rke2-server, rke2-agent
+		serviceName = fmt.Sprintf("%s-%s", product, serviceType)
+	}
+	LogLevel("debug", fmt.Sprintf("Service Name: %s", serviceName))
+	return serviceName
+}
+
+// stop k3s or rke2 service on the node
+func StopService(product string, ip string, serviceType string) (string, error) {
```

Comment:
these 3 func u can also improve your code and the framework as a whole creating something like a ManagerService . maybe something like that 
```
func ManageClusterService(action, product string, agents, servers []string) error {
	var err error

	for _, ip := range agents {
		if err = serviceCmds(action, product, ip); err != nil {
			return err
		}
	}

	for _, ip := range servers {
		if err = serviceCmds(action, product, ip); err != nil {
			return err
		}
	}

	return nil
}

func serviceCmds(action, product, ip string) error {
	var cmd string
	var err error

	switch action {
	case "start":
		cmd = fmt.Sprintf("timeout 2m sudo systemctl start %s*", product)
		_, err = RunCommandOnNode(cmd, ip)
		if err != nil {
			return err
		}
	case "stop":
		cmd = fmt.Sprintf("sudo systemctl stop %s*", product)
		_, err = RunCommandOnNode(cmd, ip)
		if err != nil {
			return err
		}
	default:
		return ReturnLogError("invalid action: %s. Must be 'start' or 'stop'", action)
	}

	return nil
}
```

Reply:
it will also remove some burden from your test cases

Reply:
+ u can also add more stuff like restart and etc

Reply:
fixed